### PR TITLE
[TS migration] Migrate 'ImageView' component to TypeScript

### PR DESCRIPTION
**Pull request by tienifr**
### Fixed Issues

$ https://github.com/Expensify/App/issues/25113
PROPOSAL: NA

### Tests

1. Open any chat that has the image
2. Click on the image 
3. Verify that image is display correctly
4. Verify that zoom in/out is working correctly

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as above

### QA Steps

1. Open any chat that has the image
2. Click on the image 
3. Verify that image is display correctly
4. Verify that zoom in/out is working correctly

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/113963320/2fa2515f-4937-4b61-a9c4-cfb386a718cb

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/113963320/d4f110d5-a9f6-4850-bd63-734d5d3e968c

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/113963320/c2aaf445-b992-4d44-bfe3-549558c14063

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/113963320/e9db0c49-b311-48eb-a869-39e358c729cd

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/113963320/cc8d7bf1-7867-4f50-aef8-350106b1bc17

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/113963320/df6af666-986d-42c6-af5c-034d8ebc907d

</details>

### CODE REVIEW

File: src/components/ImageView/index.tsx [hunk 1]
```diff
@@ -121,10 +101,8 @@ function ImageView({isAuthTokenRequired, url, fileName, onError}) {
         return {offsetX, offsetY};
     };
 
-    /**
-     * @param {SyntheticEvent} e
-     */
-    const onContainerPress = (e) => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const onContainerPress = (e: any) => {
```
Comment (s77rt) [inline src/components/ImageView/index.tsx:105]: The native event here is PointerEvent. Should we just convert the type for this case? I think we should bring this to slack

File: src/components/Lightbox.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ const propTypes = {
     activeIndex: PropTypes.number,
 
     /** Additional styles to add to the component */
-    style: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.object), PropTypes.object]),
+    style: PropTypes.oneOfType([PropTypes.bool, PropTypes.arrayOf(PropTypes.object), PropTypes.object]),
```
Comment (s77rt) [inline src/components/Lightbox.js:47]:
```suggestion
    style: stylePropTypes,
```

In `index.native.tsx` add this comment:

```js
            // @ts-expect-error TODO: Remove this once Lightbox (https://github.com/Expensify/App/pull/33081) is migrated to TypeScript.
            style={style}

```

File: src/components/MultiGestureCanvas/types.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+type ZoomRange = {
+    min: number;
+    max: number;
+};
+
+export default ZoomRange;
```
Comment (s77rt) [inline src/components/MultiGestureCanvas/types.ts:6]:
```suggestion
export type {ZoomRange};
```

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/components/ImageView/types.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    style?: StyleProp<ViewStyle>;
+
+    /** Range of zoom that can be applied to the content by pinching or double tapping. */
+    zoomRange?: ZoomRange;
+};
+
+type ImageLoadNativeEventData = {
+    width: number;
+    height: number;
+};
```
Comment (fabioh8010) [inline src/components/ImageView/types.ts:49]:
```suggestion
};

```

nit

File: src/components/ImageView/index.native.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    url,
+    onScaleChanged,
+    onPress = () => {},
+    style = {},
+    zoomRange = zoomRangeDefaultProps.zoomRange,
+    onError,
+    isUsedInCarousel,
+    isSingleCarouselItem,
+    carouselItemIndex,
+    carouselActiveItemIndex,
```
Comment (fabioh8010) [inline src/components/ImageView/index.native.tsx:17]:
```suggestion
    carouselItemIndex = 0,
    carouselActiveItemIndex = 0,
```

File: src/components/ImageView/index.native.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import Lightbox from '@components/Lightbox';
+import {zoomRangeDefaultProps} from '@components/MultiGestureCanvas/propTypes';
+import type * as ImageViewTypes from './types';
```
Comment (fabioh8010) [inline src/components/ImageView/index.native.tsx:4]:
```suggestion
import type {ImageViewProps} from './types';
```

File: src/components/ImageView/index.tsx [hunk 7]
```diff
@@ -27,40 +30,26 @@ function ImageView({isAuthTokenRequired, url, fileName, onError}) {
     const [zoomScale, setZoomScale] = useState(0);
     const [zoomDelta, setZoomDelta] = useState({offsetX: 0, offsetY: 0});
```
Comment (fabioh8010) [inline src/components/ImageView/index.tsx:31]: Please create a type to represent this data structure and use it here.

File: src/components/ImageView/index.tsx [hunk 8]
```diff
@@ -27,40 +30,26 @@ function ImageView({isAuthTokenRequired, url, fileName, onError}) {
...
         setContainerWidth(width);
     };
 
-    /**
-     * When open image, set image width, height.
-     * @param {Number} imageWidth
-     * @param {Number} imageHeight
-     */
-    const setImageRegion = (imageWidth, imageHeight) => {
+    const setImageRegion = (imageWidth: number, imageHeight: number) => {
```
Comment (fabioh8010) [inline src/components/ImageView/index.tsx:52]:
```suggestion
    /**
     * When open image, set image width, height.
     */
    const setImageRegion = (imageWidth: number, imageHeight: number) => {
```

File: src/components/ImageView/index.tsx [hunk 9]
```diff
@@ -78,32 +67,23 @@ function ImageView({isAuthTokenRequired, url, fileName, onError}) {
...
-    /**
-     * Convert touch point to zoomed point
-     * @param {Boolean} x x point when click zoom
-     * @param {Boolean} y y point when click zoom
-     * @returns {Object} converted touch point
-     */
-    const getScrollOffset = (x, y) => {
-        let offsetX;
-        let offsetY;
+    const getScrollOffset = (x: number, y: number) => {
```
Comment (fabioh8010) [inline src/components/ImageView/index.tsx:84]:
```suggestion
    /**
     * Convert touch point to zoomed point
     * @param x point when click zoom
     * @param y point when click zoom
     * @returns converted touch point
     */
    const getScrollOffset = (x: number, y: number) => {
```

File: src/components/ImageView/index.tsx [hunk 10]
```diff
@@ -148,13 +126,10 @@ function ImageView({isAuthTokenRequired, url, fileName, onError}) {
...
 
-    /**
-     * @param {SyntheticEvent} e
-     */
     const trackPointerPosition = useCallback(
-        (e) => {
+        (e: MouseEvent) => {
             // Whether the pointer is released inside the ImageView
-            const isInsideImageView = scrollableRef.current.contains(e.nativeEvent.target);
+            const isInsideImageView = scrollableRef.current?.contains(e.target as Node);
```
Comment (fabioh8010) [inline src/components/ImageView/index.tsx:132]: Please test if this change won't create regressions

File: src/components/ImageView/index.tsx [hunk 11]
```diff
@@ -78,32 +67,23 @@ function ImageView({isAuthTokenRequired, url, fileName, onError}) {
         setIsZoomed(false);
     };
 
-    const imageLoad = ({nativeEvent}) => {
+    const imageLoad = ({nativeEvent}: {nativeEvent: ImageViewTypes.ImageLoadNativeEventData}) => {
```
Comment (s77rt) [inline src/components/ImageView/index.tsx:70]: Let's use `NativeSyntheticEvent<ImageLoadNativeEventData>` on the event variable

**Comment by blazejkustra** [comment]
@tienifr Please change the title to "[TS migration] Migrate 'ImageView' component to TypeScript"

File: src/components/MultiGestureCanvas/types.ts [hunk 12]
```diff
@@ -3,4 +3,5 @@ type ZoomRange = {
     max: number;
 };
 
-export default ZoomRange;
+// eslint-disable-next-line import/prefer-default-export
+export type {ZoomRange};
```
Comment (fabioh8010) [inline src/components/MultiGestureCanvas/types.ts:7]: What's the reason for this change?

File: src/components/ImageView/index.tsx (hunk 10 above)
Comment (fabioh8010) [reply src/components/ImageView/index.tsx]: One way to solve this:

First, add this `ReactMouseEvent` type to `react` import:

```ts
import React, {MouseEvent as ReactMouseEvent, useCallback, useEffect, useRef, useState} from 'react';
```

Then, use it like this:

```ts
    const trackPointerPosition = useCallback(
        (e: MouseEvent) => {
            const mouseEvent = e as unknown as ReactMouseEvent;
            // Whether the pointer is released inside the ImageView
            const isInsideImageView = scrollableRef.current?.contains(mouseEvent.nativeEvent.target as Node);

            if (!isInsideImageView && isZoomed && isDragging && isMouseDown) {
                setIsDragging(false);
                setIsMouseDown(false);
            }
        },
        [isDragging, isMouseDown, isZoomed],
    );
```

File: src/components/ImageView/index.tsx [hunk 13]
```diff
@@ -165,14 +140,14 @@ function ImageView({isAuthTokenRequired, url, fileName, onError}) {
     );
 
     const trackMovement = useCallback(
-        (e) => {
+        (e: MouseEvent) => {
```
Comment (fabioh8010) [inline src/components/ImageView/index.tsx:155]: Same here:

```ts
    const trackMovement = useCallback(
        (e: MouseEvent) => {
            const mouseEvent = e as unknown as ReactMouseEvent;
            if (!isZoomed) {
                return;
            }

            if (isDragging && isMouseDown && scrollableRef.current) {
                const x = mouseEvent.nativeEvent.x;
                const y = mouseEvent.nativeEvent.y;
                const moveX = initialX - x;
                const moveY = initialY - y;
                scrollableRef.current.scrollLeft = initialScrollLeft + moveX;
                scrollableRef.current.scrollTop = initialScrollTop + moveY;
            }

            setIsDragging(isMouseDown);
        },
        [initialScrollLeft, initialScrollTop, initialX, initialY, isDragging, isMouseDown, isZoomed],
    );
```

**Comment by s77rt** [comment]
@tienifr Is this ready for review? I still see the `any` type being used.

File: src/components/ImageView/index.tsx (hunk 13 above)
Comment (fabioh8010) [reply src/components/ImageView/index.tsx:155]: Well if it's well tested and it's working I believe you can change to remove `nativeEvent`, same for the other case `const isInsideImageView = scrollableRef.current.contains(e.nativeEvent.target);`.

File: src/components/MultiGestureCanvas/types.ts (hunk 12 above)
Comment (fabioh8010) [reply src/components/MultiGestureCanvas/types.ts:7]: I understand, but in case we have more types we just change from default export to named ones, and do a refactor, wouldn't be such a big problem to do 🙂 I would prefer using a default export and removing this error supression

File: src/components/ImageView/index.tsx (hunk 13 above)
Comment (blazejkustra) [reply src/components/ImageView/index.tsx:155]: How about this?
```ts
    const trackMovement = useCallback(
        (event: MouseEvent) => {
            if (!isZoomed) {
                return;
            }

            if (isDragging && isMouseDown && scrollableRef.current) {
                const x = event.x;
                const y = event.y;
                const moveX = initialX - x;
                const moveY = initialY - y;
                scrollableRef.current.scrollLeft = initialScrollLeft + moveX;
                scrollableRef.current.scrollTop = initialScrollTop + moveY;
            }

            setIsDragging(isMouseDown);
        },
        [initialScrollLeft, initialScrollTop, initialX, initialY, isDragging, isMouseDown, isZoomed],
    );
```

Comment (blazejkustra) [reply src/components/ImageView/index.tsx:155]: Same for the other function: 
```ts
    const trackPointerPosition = useCallback(
        (event: MouseEvent) => {
            // Whether the pointer is released inside the ImageView
            const isInsideImageView = scrollableRef.current?.contains(event.target as Node);

            if (!isInsideImageView && isZoomed && isDragging && isMouseDown) {
                setIsDragging(false);
                setIsMouseDown(false);
            }
        },
        [isDragging, isMouseDown, isZoomed],
    );
```

**Comment by blazejkustra** [comment]
`onContainerPress` needs a bigger refactor, I think we have invalid type for onPress. Here is a [commit](https://github.com/Expensify/App/commit/b628878092163edfeadfe922c6d08e9a14d41762) where I fixed it. The idea behind it is to extract `PressableOnPress` and extend the event with `SyntheticEvent<Element, PointerEvent>`:

```ts
type PressableOnPress = (event?: GestureResponderEvent | KeyboardEvent | SyntheticEvent<Element, PointerEvent>) => void | Promise<void>;

// ImageView/index.tsx
    const onContainerPress: OnPress = (event) => {
        if (!isZoomed && !isDragging) {
            if (event && 'nativeEvent' in event && 'offsetX' in event.nativeEvent) {

```

Please test it on each platform (console log the event), to make sure the type is correct, if needed adjust it.

File: src/components/ImageView/index.tsx [hunk 14]
```diff
@@ -101,10 +113,9 @@ function ImageView({isAuthTokenRequired = false, url, fileName, onError = () =>
         return {offsetX, offsetY};
     };
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const onContainerPress = (e: any) => {
+    const onContainerPress = (e?: GestureResponderEvent | KeyboardEvent | SyntheticEvent<Element, PointerEvent>) => {
         if (!isZoomed && !isDragging) {
-            if (e.nativeEvent) {
+            if (e && 'nativeEvent' in e && 'offsetX' in e.nativeEvent) {
```
Comment (s77rt) [inline src/components/ImageView/index.tsx:118]:
```suggestion
            if (e && 'nativeEvent' in e && e.nativeEvent instanceof PointerEvent) {
```

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


android

</details>

<details>
<summary>Android: mWeb Chrome</summary>


mweb-chrome

</details>

<details>
<summary>iOS: Native</summary>


ios

</details>

<details>
<summary>iOS: mWeb Safari</summary>


mweb-safari

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


web

</details>

<details>
<summary>MacOS: Desktop</summary>


desktop

</details>

**Comment by s77rt** [comment]
@tienifr There is a failing test (Reassure Performance Tests) can you try merge main see if that fixes it?

**Review by s77rt — APPROVED** [review]

File: src/components/MultiGestureCanvas/types.ts (hunk 12 above)
Comment (mountiny) [reply src/components/MultiGestureCanvas/types.ts:7]: @tienifr did you forget to address this feedbakc?

**Review by mountiny — CHANGES_REQUESTED** [review]
@tienifr one pending comment

**Comment by mountiny** [comment]
@tienifr Can you please sync with main too and retest quickly locally?

**Review by mountiny — APPROVED** [review]
Waiting for confirmation all works after syncing with main
